web/src/components/common/layouts/NotificationMenu.vue

```diff
@@ -0,0 +1,185 @@
+            @click="markAsRead(notification)"
+          >
+            <template #append>
+              <v-btn
+                v-if="notification.href"
+                size="x-small"
+                class="mt-0 ml-2 align-top"
+                icon="mdi-link-variant"
+                :color="!notification.readAt ? 'error' : 'success'"
+                @click.stop="openNotification(notification)"
+              />
+            </template>
+            <p class="text-subtitle-1 font-weight-retular font-italic textSecondary">
+              {{ formatDate(notification.createdAt) }}
+            </p>
+            <div>
+              <h6 class="text-subtitle-1 font-weight-semibold mb-1">{{ notification.title }}</h6>
+            </div>
+            <p class="text-subtitle-1 font-weight-regular textSecondary">
+              {{ notification.subtitle }}
+            </p>
+          </v-list-item>
+          <v-divider></v-divider>
+        </v-list>
+      </div>
+      <div class="py-4 px-6 text-center">
+        <v-btn
+          variant="outlined"
+          block
+          :to="{ name: 'NotificationsPage' }"
+          text="See all Notifications"
+          @click="showMenu = false"
+        />
+      </div>
+    </v-sheet>
+  </v-menu>
+</template>
+
+<script setup lang="ts">
+import { computed, ref } from "vue"
+import { useRouter } from "vue-router"
+
+import { formatDate } from "@/utils/formatters"
+
+import notificationsApi, { type Notification } from "@/api/notifications-api"
+
+import { CURRENT_USERS_TIMEZONE } from "@/use/use-current-user"
+import useNotifications from "@/use/use-notifications"
+
+const latestNotificationsQuery = computed(() => ({
+  perPage: 5,
+}))
+
+const { notifications: latestNotifications, refresh: refreshLatestNotifications } =
+  useNotifications(latestNotificationsQuery)
+
+const unreadNotificationsQuery = computed(() => ({
+  where: {
+    readAt: null,
+  },
+  perPage: 1, // we only care about the total count
+}))
+
+const { totalCount: totalCountUnreadNotifications, refresh: refreshUnreadNotifications } =
+  useNotifications(unreadNotificationsQuery)
+const hasUnreadNotifications = computed(() => totalCountUnreadNotifications.value > 0)
+
+const todaysNotificationsQuery = computed(() => ({
+  filters: {
+    createdTodayInUserTimezone: CURRENT_USERS_TIMEZONE,
+  },
+  perPage: 1, // we only care about the total count
+}))
+
+const { totalCount: totalCountTodaysNotifications, refresh: refreshTodaysNotifications } =
+  useNotifications(todaysNotificationsQuery)
+
+async function refresh() {
+  await refreshLatestNotifications()
+  await refreshUnreadNotifications()
+  await refreshTodaysNotifications()
+}
+
+async function markAsRead(notification: Notification) {
+  if (notification.readAt) return
+
+  try {
+    const { notification: updatedNotification } = await notificationsApi.update(notification.id, {
+      readAt: new Date().toISOString(),
```

Comment:
Timestamp values shouldn't be set in the front-end from a data security perspective.
And while I see that this value is overriden in the back-end, so isn't technically creating a security problem, it's instead creating a conceptual and readability problem.

I'd like to fix this via a state change controller for setting the "read" state.